### DISCUSSION TITLE
Is it possible to show suspense fallback for nested lazy-loaded routes?

### DISCUSSION THREAD

**Discussion by misha-erm** [remix-run/remix#9879] [Q&A]
Hello, 

I've noticed that remix blocks navigation until the code for nested route is loaded. Is there a way to somehow defer lazy-loaded code and immediately show suspense fallback similar to how we can defer loader's data? I tried wrapping `<Outlet/>` with suspense but that doesn't seem to work

Thanks in advance for any tips

here is a minimal repro: https://github.com/misha-erm/remix-nav-block-report

https://github.com/user-attachments/assets/0cbb2420-8227-450b-b014-9c60a444b45f

**Comment by sergiodxa** [comment]
You can use the useNavigation hook to know a navigation is pending and render a pending ui, but route navigation is not using suspense right now so you can't use Suspense for that.

**Reply by misha-erm** [reply]
Thanks for the answer 🙇🏻 
Are there any plans for supporting it? Maybe in RR7 ?
